pkg/server/api/agent/v1/service.go

```diff
@@ -283,21 +283,26 @@ func (s *Service) AttestAgent(stream agentv1.Agent_AttestAgentServer) error {
 		}
 	}
 
-	agentID := attestResult.AgentID
-	log = log.WithField(telemetry.AgentID, agentID)
-	rpccontext.AddRPCAuditFields(ctx, logrus.Fields{telemetry.AgentID: attestResult.AgentID})
-
-	if err := idutil.CheckAgentIDStringNormalization(agentID); err != nil {
-		return api.MakeErr(log, codes.Internal, "agent ID is malformed", err)
-	}
-
-	agentSpiffeID, err := spiffeid.FromString(agentID)
+	// Ideally we'd do stronger validation that the ID is within the Node
+	// Attestors scoped area of the reserved agent namespace, but historically
+	// we haven't been strict here and there are deployments that are emitting
+	// such IDs.
+	// TODO: Enforce that IDs produced by Node Attestors are in the reserved
+	// namesepace for that Node Attestor.
+	//     e.g. spiffe://<trustdomain>/spire/agent/<nodeattestor>/<remainder>
+	agentID, err := spiffeid.FromString(attestResult.AgentID)
 	if err != nil {
-		return api.MakeErr(log, codes.Internal, "invalid agent id", err)
+		return api.MakeErr(log, codes.Internal, "invalid agent ID", err)
+	}
+	if agentID.Path() == idutil.ServerIDPath {
+		return api.MakeErr(log, codes.Internal, "agent ID cannot collide with the server ID", nil)
 	}
 
+	log = log.WithField(telemetry.AgentID, agentID)
+	rpccontext.AddRPCAuditFields(ctx, logrus.Fields{telemetry.AgentID: attestResult.AgentID})
```

Comment:
why attestResult.AgentID} is used for audit log, but agentID on regular log?

Reply:
good catch. Missed one!